code/game/atoms.dm

```diff
@@ -381,3 +367,20 @@ var/list/blood_splatter_icons = list()
 //the sight changes to give to the mob whose perspective is set to that atom (e.g. A mob with nightvision loses its nightvision while looking through a normal camera)
 /atom/proc/update_remote_sight(mob/living/user)
 	return
+
+/atom/proc/add_vomit_floor(mob/living/carbon/M, toxvomit = 0)
+	if(istype(src,/turf/simulated) )
+		var/obj/effect/decal/cleanable/vomit/V = PoolOrNew(/obj/effect/decal/cleanable/vomit, src)
+		// Make toxins vomit look different
+		if(toxvomit)
+			V.icon_state = "vomittox_[pick(1,4)]"
+		if(M.reagents)
+			clear_reagents_to_vomit_pool(M,V)
+
+/atom/proc/clear_reagents_to_vomit_pool(mob/living/carbon/M, obj/effect/decal/cleanable/vomit/V)
+	M.reagents.trans_to(V, M.reagents.total_volume / 10)
+	for(var/datum/reagent/R in reagents.reagent_list)                //clears the stomach of anything that might be digested as food

```

Comment:
When a floor turf tries to call clear_reagents_to_vomit_pool, it encounters an error that stops the proc. The turf's "reagents" var is null, so when you ask it to do use "reagents.reagent_list" and so it shits itself, figuratively.